Treat ids with double primes correctly (fixes #11)

aztek opened the pull request:
Do not highlight identifiers with double primes as character literals in Haskell. The bug seems to be in an extra condition that checks if we are at the beginning of a character literal. The test file for Haskell looks fine after this change.

aztek committed:
Treat ids with double primes correctly (fixes #11)

Do not highlight identifiers with double primes as character literals in Haskell. The bug seems to be in an extra condition that checks if we are at the beginning of a character literal. The test file for Haskell looks fine after this change.

tmont commented:
Cool! Can you provide a little haskell snippet containing a double prime that we can put in the test file?

aztek committed:
Add test snippet to the Haskell test file

Extra code to test identifiers with single quotes in them. Only 'c' should be highlighted as a character literal, the rest are legal identifiers.

aztek commented:
This one should be enough.

``` haskell
c :: Char
c = c'
  where c' = c''
        c'' = c'''
        c''' = c'c'
        c'c' = c'c''
        c'c'' = c''c''
        c''c'' = 'c'
```

(Apparently, GitHub's syntax highlighter doesn't handle quotes wery well either.)
